internal/resources/kustomization/resource_kustomization_test.go

```diff
@@ -64,12 +66,61 @@ func testGetDefaultAcceptanceConfig(t *testing.T) *testAcceptanceConfig {
 		GitRepositoryResource:     gitRepositoryResource,
 		GitRepositoryResourceVar:  gitRepositoryResourceVar,
 		GitRepositoryName:         acctest.RandomWithPrefix(gitRepositoryNamePrefix),
+		Namespace:                 "tanzu-continuousdelivery-resources",
 	}
 }
 
+func getConfigureContextFunc() func(_ context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
+	if _, found := os.LookupEnv("ENABLE_KUSTOMIZATION_ENV_TEST"); !found {
+		return authctx.ProviderConfigureContextWithDefaultTransportForTesting
+	}
+
+	return authctx.ProviderConfigureContext
+}
+
+func getSetupConfig(config *authctx.TanzuContext) error {
+	if _, found := os.LookupEnv("ENABLE_KUSTOMIZATION_ENV_TEST"); !found {
+		return config.SetupWithDefaultTransportForTesting()
+	}
+
+	return config.Setup()
+}
+
 func TestAcceptanceForKustomizationResource(t *testing.T) {
 	testConfig := testGetDefaultAcceptanceConfig(t)
 
+	// If the flag to execute kubernetes secret tests is not found, run this as a unit test by setting up an http intercept for each endpoint.
+	_, found := os.LookupEnv("ENABLE_KUSTOMIZATION_ENV_TEST")
+	if !found {
+		os.Setenv("TF_ACC", "true")
+		// os.Setenv("TMC_ENDPOINT", "playground.tmc.cloud.vmware.com")
```

Comment:
Nit: remove commented lines

Reply:
updated PTAL

Reply:
updated PTAL

Reply:
I've also added the PUT responder so please check